src/App.tsx

```diff
@@ -10,13 +11,15 @@ import './App.css';
 
 const App = () => {
   return (
-    <Router basename={'/perf-basecamp'}>
+    <Router>
       <NavBar />
-      <Routes>
-        <Route path="/" element={<Home />} />
-        <Route path="/search" element={<Search />} />
-      </Routes>
-      <Footer />
+      <Suspense fallback={<div>Loading...</div>}>
+        <Routes>
+          <Route path="/" element={<Home />} />
+          <Route path="/search" element={<Search />} />
+        </Routes>
+        <Footer />
+      </Suspense>
```

Comment:
Routes만 Suspense로 감싸면 레이아웃 변경이 발생해서 Footer를 Suspense 안에 넣으셨군요 👍

---

Routes만 Suspense로 감싸도 레이아웃 변경이 일어나지 않게 하기 위해 찾아봤던 내용을 공유드려요!

**스켈레톤 UI**를 만들어서 **Suspense의 fallback이 앞으로 로딩될 컴포넌트의 영역을 미리 알고 자리를 차지하게 만들면** Footer 컴포넌트가 Suspense 밖에 위치해도 괜찮을 것 같아요!

 반영하라는 건 아니고 이렇게도 할 수 있다는 걸 공유드리고 싶었어요 🙇🏻‍♂️

```suggestion
<Routes>
        <Route
          path="/"
          element={
            <Suspense fallback={<HomeSkeleton />}>
              <Home />
            </Suspense>
          }
        />
        <Route
          path="/search"
          element={
            <Suspense fallback={<SearchSkeleton />}>
              <Search />
            </Suspense>
          }
        />
      </Routes>
```

Reply:
> Routes만 Suspense로 감싸면 레이아웃 변경이 발생해서 Footer를 Suspense 안에 넣으셨군요 👍

네 맞아요! Footer 태그의 layout shift 를 막고자 Suspense 내부로 이동했었어요
(Suspense 내부에 존재하지 않을 때 아래와 같이 performance 탭에서 확인하실 수 있어요)
<img width="1859" height="1217" alt="image" src="https://github.com/user-attachments/assets/069cfd48-2c5d-4aac-8312-736da62d227b" />

Footer를 Suspense로 감싸면 메인 콘텐츠가 로딩될 때 Footer도 함께 사라졌다가 나타나게 되어서 적절한 layout shift 해결방안은 아니라고 생각했는데 로건이 말해준대로 스켈레톤 ui를 적용하는 것도 방법이겠네요!
스켈레톤 UI가 placeholder 와 같은 역할을 하는군요 ~! 하나 배워갑니다 ㅎㅎ 🙇‍♂️🙇‍♂️